storage2/src/snapshot.rs

```diff
@@ -28,3 +37,50 @@ impl Snapshot {
         self.jmt_version
     }
 }
+
+/// A reader interface for rocksdb. NOTE: it is up to the caller to ensure consistency between the
+/// rocksdb::DB handle and any write batches that may be applied through the writer interface.
+impl TreeReader for Snapshot {
+    /// Gets node given a node key. Returns `None` if the node does not exist.
+    fn get_node_option(&self, node_key: &NodeKey) -> Result<Option<Node>> {
+        let node_key = node_key.clone();
```

Comment:
I don't think the node key needs to be cloned. It's only used by reference below.